File: java_projects/glue_code/commons-graph/src/main/java/org/apache/commons/graph/shortestpath/PathSourceSelector.java
package org.apache.commons.graph.shortestpath;

import org.graalvm.polyglot.Value;
import org.graalvm.polyglot.PolyglotException;
import org.apache.commons.graph.ContextInitializer;
import org.apache.commons.graph.ExceptionHandler;
import org.apache.commons.graph.IntegrationUtils;
import org.apache.commons.graph.weight.OrderedMonoid;
public interface PathSourceSelector<V, WE, W> {
    <H extends V> TargetSourceSelector<V, WE, W> from(H source);
    <WO extends OrderedMonoid<W>> AllVertexPairsShortestPath<V, WE, W> applyingFloydWarshall(
;}

File: data/skeletons/commons-graph/src/main/org/apache/commons/graph/shortestpath/PathSourceSelector.py
# Imports Begin
from src.main.org.apache.commons.graph.weight.OrderedMonoid import *
from src.main.org.apache.commons.graph.shortestpath.TargetSourceSelector import *
from src.main.org.apache.commons.graph.shortestpath.AllVertexPairsShortestPath import *
import typing
import pathlib
from abc import ABC

# Imports End


class PathSourceSelector(ABC):

    # Class Fields Begin
    # Class Fields End

    # Class Methods Begin
    def from_(
        self, source: typing.Any
    ) -> TargetSourceSelector[typing.Any, typing.Any, typing.Any]:
        pass

    def applyingFloydWarshall(
        self, weightOperations: typing.Any
    ) -> AllVertexPairsShortestPath[typing.Any, typing.Any, typing.Any]:
        pass

    # Class Methods End
